crashreports/internal/win32/CrashReportContext.cpp

```diff
@@ -0,0 +1,165 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ CrashReportContext.cpp
+
+ Vitaly Sverchinsky
+
+ Some parts of the code are designed to operate while app is crashing,
+ so there may be some restrictions on heap usage. For more information
+ please read Breakpad documentation.
+
+ **********************************************************************/
+
+#include "CrashReportContext.h"
+
+#include <locale>
+#include <codecvt>
+#include <sstream>
+#include "client/windows/handler/exception_handler.h"
+
+namespace
+{
+	//copy src(null-terminated) to dst, 
+	//returns false if dest is not large enough
+	bool StrcpyChecked(wchar_t* dest, size_t destsz, const wchar_t* src)
+	{
+		auto len = wcslen(src);
+		if (len < destsz)
+		{
+			memcpy(dest, src, sizeof(wchar_t) * len);
+			dest[len] = 0;
+			return true;
+		}
+		return false;
+	}
+
+	//appends src(null-terminated) to dest, destsz is the total dest buffer size, not remaining
+	//returns false if dest is not large enough
+	bool StrcatChecked(wchar_t* dest, size_t destsz, const wchar_t* src)
+	{
+		auto srclen = wcslen(src);
+		auto dstlen = wcslen(dest);
+		if (srclen + dstlen < destsz)
+		{
+			memcpy(dest + dstlen, src, sizeof(wchar_t) * srclen);
+			dest[srclen + dstlen] = 0;
+			return true;
+		}
+		return false;
+	}
+
+	//converts parameters map to a string, so that the Crash Reporting program
+	//would understand them
+	std::string StringifyParameters(const std::map<std::string, std::string>& parameters)
+	{
+		std::stringstream stream;
+
+		std::size_t parameterIndex = 0;
+		std::size_t parametersCount = parameters.size();
+		for (auto& pair : parameters)
+		{
+			stream << pair.first.c_str() << "=\\\"" << pair.second.c_str() << "\\\"";
+			++parameterIndex;
```

Comment:
Is there a difference between shell programs on different platforms explaining why you need the extra \ characters in stream here and not on unix?